### PR TITLE
Implement queue operations for AWS SQS

### LINKED ISSUE #114: Create AWS SQS client
Add implementation for aws sqs

### PR DESCRIPTION
**Pull request by adelinag08**
Closes #114

### CODE REVIEW

File: src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    /// <inheritdoc />
+    public Source<QueueElement, NotUsed> GetQueueMessages(string queueName, TimeSpan visibilityTimeout,
+        int prefetchCount, TimeSpan pollInterval)
+    {
+        this.logger.LogDebug("Creating a stream from queue: {queueName}, using visibility timeout {visibilityTimeout}",
+            queueName, visibilityTimeout);
+        var settings = SqsSourceSettings.Default
+            .WithVisibilityTimeout(visibilityTimeout)
+            .WithMaxBatchSize(Math.Min(10, prefetchCount))
```
Comment (s-vitaliy) [inline src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs:64]: Why do you set a minimum value here?

File: src/SnD.Sdk/Storage/Base/IQueueService.cs [hunk 2]
```diff
@@ -3,13 +3,16 @@
...
+using Snd.Sdk.Storage.Models.Base;
 
 namespace Snd.Sdk.Storage.Base
 {
     /// <summary>
     /// Remote object queue abstractions
     /// </summary>
-    public interface IQueueService
+    public interface IQueueService<TSendResponse, TReleaseResponse>
+        where TSendResponse : IMessageSendResponse
```
Comment (s-vitaliy) [inline src/SnD.Sdk/Storage/Base/IQueueService.cs:14]: I think you don't need type bounds here

File: src/SnD.Sdk/Storage/Models/Base/IMessageReleaseResponse.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+namespace Snd.Sdk.Storage.Models.Base;
+
+/// <summary>
+/// Type-safe abstraction of different queue response from releasing a message.
+/// </summary>
+public class IMessageReleaseResponse
```
Comment (s-vitaliy) [inline src/SnD.Sdk/Storage/Models/Base/IMessageReleaseResponse.cs:6]: Class names should not start with `I`

**Review by s-vitaliy — APPROVED** [review]
LGTM in general, but need some more work to do

File: src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs (hunk 1 above)
Comment (adelinag08) [reply src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs:64]: so I avoid having a value larger than 10, 10 being the maximum allowed

File: src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    /// </summary>
+    /// <param name="client">The Amazon SQS client.</param>
+    /// <param name="logger">The logger instance.</param>
+    public AmazonSqsService(IAmazonSQS client, ILogger<AmazonSqsService> logger)
+    {
+        this.client = client;
+        this.logger = logger;
+    }
+
+    private string GetQueueUrlAsync(string queueName)
```
[the remark this answers is not in the document]
Comment (adelinag08) [reply src/SnD.Sdk/Storage/Amazon/AmazonSqsService.cs]: fixed, made it async and called `GetAwaiter().GetResult()` where I needed the value

**Pull request merged**